### ISSUE TITLE
TypeVar are handled inconsistency when in-lined compared to in TypeAlias

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#3272]
**Describe the bug**
TypeVar seem to behave differently when inlined compared to when used in a TypeAlias (not sure whether it is a TypeAlias).

**To Reproduce**
```py
from typing import Hashable, Mapping, TypeAlias, TypeVar

HashableT = TypeVar("HashableT", bound=Hashable)
Renamer: TypeAlias = Mapping[HashableT, Hashable]  # should allow covariant keys

class DataFrame:
    def rename(self, mapper: Renamer) -> None:
        ...

    def rename2(self, mapper: Mapping[HashableT, Hashable]) -> None:
        ...

df = DataFrame()
col_map = {"a": "b"}
df.rename(col_map)  # error
df.rename2(col_map)  # no error
```
> error: Argument of type "dict[str, str]" cannot be assigned to parameter "mapper" of type "Renamer[Unknown]" in function "rename"
>     TypeVar "_KT@Mapping" is invariant
>       Type "str" cannot be assigned to type "HashableT@Renamer" (reportGeneralTypeIssues)

**Expected behavior**
Either both calls error or none.

**Comment by JelleZijlstra** [comment]
This is expected. The type alias automatically becomes a generic alias, and if you don't provide type arguments they are assumed to be Any. You need to rewrite `Renamer[HashableT]` instead.

**Comment by erictraut** [comment]
This is expected behavior. You're defining a generic type alias called `Renamer`. It accepts a single type argument corresponding to the type parameter `HashableT`. But then you're failing to provide a type argument when you use the type alias, so pyright (by rules of PEP 484) assumes that you mean `Renamer[Any]`. If you want it to remain generic, you need to specify `Renamer[HashtableT]`.

If you enable the `reportMissingTypeArgument` diagnostic check, it will report cases where you omit the type arguments.

**Comment by twoertwein** [comment]
Thank you, both of you! It sounds very obvious after you pointed it out :)
